Packs/SentinelOne/Integrations/SentinelOne-V2/SentinelOne-V2.py

```diff
@@ -2864,171 +3212,453 @@ def extract_sentinelone_zip_file(zip_file_data: bytes, password: str) -> Tuple[s
+        demisto.debug("Successfully updated the threat status and marked as resolved")
+        note = "XSOAR - Marked as resolved \n" + delta.get("closeNotes", "")
+        client.write_threat_note_request(threat_ids=argToList(threat_id), note=note)
+    else:
+        demisto.debug("Unable to mark as resolved")
+    return ""
+
+
+def update_remote_system_command(client: Client, args: dict) -> str:
+    """update-remote-system command: pushes local changes to the remote system
+
+    :type client: ``Client``
+    :param client: XSOAR client to use
+
+    :type args: ``Dict[str, Any]``
+    :param args:
+        all command arguments, usually passed from ``demisto.args()``.
+        ``args['data']`` the data to send to the remote system
+        ``args['entries']`` the entries to send to the remote system
+        ``args['incidentChanged']`` boolean telling us if the local incident indeed changed or not
+        ``args['remoteId']`` the remote incident id
+        args: A dictionary containing the data regarding a modified incident, including: data, entries, incident_changed,
+         remote_incident_id, inc_status, delta
+
+    :return:
+        ``str`` containing the remote incident id - really important if the incident is newly created remotely
+
+    :rtype: ``str``
+    """
+    parsed_args = UpdateRemoteSystemArgs(args)
+    delta = parsed_args.delta
+    remote_incident_id = parsed_args.remote_incident_id
+    demisto.debug(f'Got the following data {parsed_args.data}, and delta {delta}.')
+    if delta:
+        demisto.debug(f'Got the following delta keys {list(delta.keys())}.')
+    demisto.debug(
+        f"Sending incident with remote ID [{parsed_args.remote_incident_id}] to remote system\n"
+    )
+    try:
+        if parsed_args.incident_changed:
+            if delta.get("closeNotes", None):
+                close_remote_incident(client, remote_incident_id, delta)
+            else:
+                update_remote_incident(client, remote_incident_id, delta)
+    except Exception as e:
+        demisto.error(f'Error in SentinelOne outgoing mirror for incident {remote_incident_id}. '
+                      f'Error message: {str(e)}')
+
+    return remote_incident_id
+
+
+def set_xsoar_incident_entries(mirrored_object: dict, entries: list, remote_incident_id: str):
+    demisto.debug("with in the set xsoar incident entries method")
+    if mirrored_object.get("threatInfo", {}).get("incidentStatus") == "resolved":
+        demisto.debug(f"Incident is closed: {remote_incident_id}")
+        entries.append(
+            {
+                "Type": EntryType.NOTE,
+                "Contents": {
+                    "dbotIncidentClose": True,
+                    "closeReason": "Incident was closed on SentinelOne",
+                },
+                "ContentsFormat": EntryFormat.JSON,
+            }
+        )
+        return entries
+    elif mirrored_object.get("threatInfo", {}).get("incidentStatus") in (
```

Comment:
Does this always say that the incident is reopened? The status can't move between `unresolved` and `in_progress`?
I think we need to find a better way to find whether the remote incident is indeed reopened, instead of reopening it every time something changes in the incident.

Reply:
We mirrored the status of the threat/incident from SentinelOne to XSOAR, even with status there are exactly 3 statuses that correspond to one another.
With mirroring it made sense to us, to move the status for incoming changes to be reflected exacted how they are on SentinelOne's side.
To check if the status was changed to/from "resolved" would be a little difficult as we would have to make a number of API calls to check that from SentinelOne.
Please let me know if this makes sense, or should be try to check if the status changed to/from "resolved" and we only take action on those changes, thanks

Reply:
> Here we tried to achieve the exact sync between the remote(SentinelOne) and XSOAR platforms so we are reopening it whenever the status is other than "resolved". If you think this is not a good approach to do then we will remove that entire reopening of the incident mechanism from the code

Let's discuss this in the demo next week.

Reply:
I've looked at it again- it is fine to reopen it in the way you do, but it is also essential to set the status to the actual status of the incident (meaning to mirror `threatInfo.incidentStatus`, but if `threatInfo.incidentStatusDescription` does that it is also fine).
Also, reopen the incident only if `close_xsoar_incident` is set to True (like you do when closing).

Reply:
we added the `close_xsoar_incident` to validate, when we made this `True` then only we close the mirrored incident on the XSOAR right? If I use this same `close_xsoar_incident` for the reopening then this will confuse the user, which means whenever we made the status updates unresolved or in_progress when  `close_xsoar_incident` is `Flase` then it's not gonna update the threat status of the mirrored incident that's not expected behavior. am I correct?

Reply:
We are only talking about reopening in XSOAR. The field that you are talking about is the `SentinelOne Threat Status`, that is mirrored with no regards to whether the `close_xsoar_incident` is checked or not (it is done by the `mirrored_data` you are sending to `GetRemoteDataResponse`, and not by the `entries` that you are setting in this function). Incident's status is changed only when `close_xsoar_incident` is `True`, so when it is `False` it will not be closed and therefor will not need to be reopened.